snakes-and-ladder/main.cpp

```diff
@@ -0,0 +1,178 @@
+/*
+ SNAKE AND LADDER GAME
+ ---------------------
+ 
+ - Works well with more than 2 players, the game continues until only one player is left
+ - Multiple 6's not handled till now
+ - unit tests not written
+ - snakes and ladders not created programitiacally
+ 
+ ** Data Structures used
+ - snakes : a map to hold the haed and tail of each snake
+ - ladders : a map to hold the top and bootom of each ladder
+ - playes : a vector of pairs where each pair contains the name of the player and it's position the board.
+ - s : no. of snakes
+ - l : no. of ladders
+ - p : no. of players
+
+ ** Functions descriptions
+ - getSnakes(): initializes the snakes
+ - getLadders(): initializes the ladders
+ - getPlayers(): gets the details of playes, initializes their initial position to zero
+ - game(): the main game function
+ 
+ 
+ */
+
+
+
+# include <iostream>
+# include <vector>
+# include <map>
+# include <cstring>
+# include <ctime>
+
+using namespace std;
+
+void getSnakes(map <int, int> &);
+void getLadders(map <int, int> &, map <int, int>);
+void getPlayers(vector <pair <string, int> > &);
+void game(int , int , int , map <int, int> , map <int, int> , vector <pair <string, int> > );
+
+
+
+int main(){
+    srand(time(NULL));
+    
+    // input the snakes co-ordinates
+    map <int, int> snakes;
+    getSnakes(snakes);
+    
+    // input the ladders co-ordinates
+    map <int, int> ladders;
+    getLadders(ladders, snakes);
+    
+    // input the names of the players and initialize their initial position as 0;
+    vector <pair <string, int> > players;
+    getPlayers(players);
+    
+    
+    // code for the game
+    game(snakes.size(), ladders.size(), players.size(), snakes, ladders, players);
+    
+    return 0;
+}
+
+
+void getSnakes(map <int, int> &snakes){
+    int s;
+    cin>>s;
+    for (int i = 0; i < s; i++){
+        int head, tail;
+        cin>>head>>tail;
+        
+        if (head <= tail){
+            cout<<"Head has to be greater than tail. WRONG INPUT!! TERMINATING!!"<<endl;
```

Comment:
It was mentioned that you can assume that head will be greater than tail. No need to validate.